src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/prisonersearch/PrisonerSearchService.kt

```diff
@@ -13,24 +14,40 @@ private val log = KotlinLogging.logger {}
 class PrisonerSearchService(private val prisonerSearchClient: PrisonerSearchClient) {
   fun lookupPrisoner(barcodeRecipient: BarcodeRecipient) {
     val prisonerSearchRequest = prisonerSearchRequest(barcodeRecipient)
-    try {
-      val matchPrisonersResponse = prisonerSearchClient.matchPrisoners(prisonerSearchRequest)
-      log.info { "Found ${matchPrisonersResponse.matches.size} potential prisoner matches" }
-    } catch (webClientResponseException: WebClientResponseException) {
-      if (webClientResponseException.statusCode == HttpStatus.FORBIDDEN) {
-        log.info { "Client token does not have correct role to call prisoner-search" }
-      } else {
-        log.error { "Failed to call prisoner-search - statusCode ${webClientResponseException.statusCode}" }
+    prisonerSearchClient.matchPrisoners(prisonerSearchRequest)
+      .onErrorResume {
+        with(it) {
+          if (this is WebClientResponseException) {
+            if (statusCode == HttpStatus.FORBIDDEN) {
+              log.info { "Client token does not have correct role to call prisoner-search" }
+            } else {
+              log.error { "Failed to call prisoner-search - statusCode $statusCode" }
+            }
+          } else {
+            log.error("Failed to call prisoner-search", it)
+          }
+        }
+          .let { Mono.empty() }
+      }
+      // In theory this is a possible way to submit the request on a different thread ... but it can't get the auth token because
+      // `userContext` is on the main request thread, so [WebClientConfiguration#addAuthHeaderFilterFunction] results in a NPE
+      // Needs some thought if we want to do this .... (other concurrency approaches may suffer the same fate if they use a different thread)
+      // .subscribeOn(Schedulers.parallel())
+      .subscribe {
+        log.info { "Found ${it.matches.size} potential prisoner matches" }
       }
-    } catch (exception: Exception) {
-      log.error("Failed to call prisoner-search", exception)
-    }
   }
+}
 
-  private fun prisonerSearchRequest(barcodeRecipient: BarcodeRecipient): PrisonerSearchRequest {
-    val names = barcodeRecipient.name.split(" ")
-    val firstName: String = names[0]
-    val lastName: String? = if (names.size > 1) names.stream().skip(1).collect(Collectors.joining(" ")) else null
-    return PrisonerSearchRequest(barcodeRecipient.prisonNumber, firstName, lastName, barcodeRecipient.dob)
-  }
+fun prisonerSearchRequest(barcodeRecipient: BarcodeRecipient): PrisonerSearchRequest {
```

Comment:
There are no unit tests that test this function, but basically names are split on space, where the first element is mapped to the first name, and all other elements are joined to become the last name .... unless there is only 1 element in which case it is mapped to the lastname and the firstname is mapped as null.